src/hooks/useMapBackground.ts

```diff
@@ -0,0 +1,16 @@
+import { DEFAULT_MAP_BACKGROUND } from "settings";
+
+import { useCheckImage } from "hooks/useCheckImage";
+
+export const useMapBackground = (
+  mapBackgroundUrl: string | undefined
```

Comment:
```suggestion
  mapBackgroundUrl?: string
```

Reply:
I'd argue that is not the same: if a prop (or parameter of a function) is optional it means you may not pass it at all. If it's `type | undefined` it means that you have to pass it but it may be undefined. 
The function needs `mapBackgroundUrl` prop, but it may be `undefined`. That's not the same as invoking this function without that parameter at all.

Reply:
Fair enough.